RecoBTag/ONNXRuntime/plugins/ParticleNetSonicJetTagsProducer.cc

```diff
@@ -0,0 +1,317 @@
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/Framework/interface/makeRefToBaseProdFrom.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/BTauReco/interface/JetTag.h"
+
+#include "DataFormats/BTauReco/interface/DeepBoostedJetTagInfo.h"
+
+#include "HeterogeneousCore/SonicTriton/interface/TritonEDProducer.h"
+
+#include "RecoBTag/FeatureTools/interface/deep_helpers.h"
+
+#include <iostream>
+#include <fstream>
+#include <algorithm>
+#include <numeric>
+#include <nlohmann/json.hpp>
+
+using namespace btagbtvdeep;
+
+class ParticleNetSonicJetTagsProducer : public TritonEDProducer<> {
+public:
+  explicit ParticleNetSonicJetTagsProducer(const edm::ParameterSet &);
+  ~ParticleNetSonicJetTagsProducer() override;
+
+  void acquire(edm::Event const &iEvent, edm::EventSetup const &iSetup, Input &iInput) override;
+  void produce(edm::Event &iEvent, edm::EventSetup const &iSetup, Output const &iOutput) override;
+  static void fillDescriptions(edm::ConfigurationDescriptions &);
+
+private:
+  typedef std::vector<reco::DeepBoostedJetTagInfo> TagInfoCollection;
+  typedef reco::JetTagCollection JetTagCollection;
+
+  std::vector<float> center_norm_pad(const std::vector<float> &input,
+                                     float center,
+                                     float scale,
+                                     unsigned min_length,
+                                     unsigned max_length,
+                                     float pad_value = 0,
+                                     float replace_inf_value = 0,
+                                     float min = 0,
+                                     float max = -1);
+  void make_inputs(const reco::DeepBoostedJetTagInfo &taginfo);
+
+  const edm::EDGetTokenT<TagInfoCollection> src_;
+  std::vector<std::string> flav_names_;             // names of the output scores
+  std::vector<std::string> input_names_;            // names of each input group - the ordering is important!
+  std::vector<std::vector<int64_t>> input_shapes_;  // shapes of each input group (-1 for dynamic axis)
+  std::vector<unsigned> input_sizes_;               // total length of each input vector
+  std::unordered_map<std::string, PreprocessParams> prep_info_map_;  // preprocessing info for each input group
+  unsigned batchSize_;
+  bool debug_ = false;
+};
+
+ParticleNetSonicJetTagsProducer::ParticleNetSonicJetTagsProducer(const edm::ParameterSet &iConfig)
+    : TritonEDProducer<>(iConfig, "ParticleNetSonicJetTagsProducer"),
+      src_(consumes<TagInfoCollection>(iConfig.getParameter<edm::InputTag>("src"))),
+      flav_names_(iConfig.getParameter<std::vector<std::string>>("flav_names")),
+      batchSize_(iConfig.getParameter<unsigned>("batchSize")),
```

Comment:
this probably shouldn't be set in the python config, but inferred directly from the jet collection in `acquire()`

Reply:
Yes, this is changed in the new version

Reply:
delete